frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -1134,14 +1123,20 @@ export default ({
       }
 
       if (toChatRoomId !== fromChatRoomId) {
+        this.ephemeral.onChatScroll?.flush()
         this.initializeState(true)
         this.ephemeral.messagesInitiated = false
         this.ephemeral.scrolledDistance = 0
-        if (sbp('chelonia/contract/isSyncing', toChatRoomId)) {
-          toIsJoined && sbp('chelonia/queueInvocation', toChatRoomId, () => initAfterSynced(toChatRoomId))
-        } else {
-          this.refreshContent()
-        }
+        // Force 'chelonia/contract/isSyncing' to be a Promise
+        Promise.resolve(sbp('chelonia/contract/isSyncing', toChatRoomId)).then((isSyncing) => {
```

Comment:
Why? (Can the comment explain?)

Reply:
Because it's easier than to write an `if`?

Reply:
Disagree, it's way more straightforward and idiomatic to not introduce any promises here and leave the code as-is.

If that's the only reason the change was made, please revert. The implication would be that all simple `if` statements in the project would need to be changed to Promises, and that's bonkers.

Reply:
Sorry, maybe my response was too simple and some meaning was lost.

> The implication would be that all simple `if` statements in the project

Not at all. The implication is that:

`sbp('chelonia/...')` will _always_ return a `Promise` if Chelonia is running in a SW, but if it's _not_ running in a SW, `sbp('chelonia/...')` may or may not return a promise.

This is handling both possibilities (should we implement a SW-less version of the app in a later commit), and in that case:

`if (foo.then) foo.then(bar) else bar(foo)` is more complex, verbose and error-prone than coercing foo into a `Promise`.

Reply:
I see... OK then couldn't you rewrite this using an `await` to keep the `if` structure? It's clearer then also what's happening with this star selector stuff (whereas with this method it's not clear to me that the star selector is the reason for this weird way of writing things).

E.g. `const isSyncing = await sbp('chelonia/contract/isSyncing', ...)`

Reply:
I'd have done it that way, but I'm not sure it's a good idea in this case, since I believe Vue expects the `watch` functions to be sync. Having async functions here could cause unexpected issues due to potential race conditions.

Reply:
OK, then just a comment explaining the reason for the weirdness (reminding about star selector) would be sufficient